### PR TITLE
[#1295] improvement(CI): separate trino image for CI and users

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Remove the Gravitino connector from the Trino CI image.

### Why are the changes needed?

In CI, we will mount the Gravitino connector to the Trino CI image to test the latest code base, so we can fully test our PR. 

Fix: #1295

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

File: dev/docker/trino/Dockerfile [hunk 1]
```diff
@@ -45,11 +45,17 @@ RUN rm -rf /usr/lib/trino/plugin/accumulo  \
...
     && rm -rf /usr/lib/trino/plugin/teradata-functions
-
 COPY --chown=trino:trino packages/trino/conf /etc/trino
-COPY --chown=trino:trino packages/gravitino-trino-connector/mysql-connector-java-8.0.27.jar /usr/lib/trino/plugin/iceberg/
-COPY --chown=trino:trino packages/gravitino-trino-connector/postgresql-42.7.0.jar /usr/lib/trino/plugin/iceberg/
-COPY --chown=trino:trino packages/gravitino-trino-connector /usr/lib/trino/plugin/gravitino
+
+ARG IMAGE_NAME
+RUN if [ "$IMAGE_NAME" = "datastrato/trino" ] ; then echo "Copying files for ${IMAGE_NAME} image" \
+  && COPY --chown=trino:trino packages/gravitino-trino-connector/mysql-connector-java-8.0.27.jar /usr/lib/trino/plugin/iceberg/ \
```
Comment (qqqttt123) [inline dev/docker/trino/Dockerfile:52]: MySQL and PostgreSQL should be copied for both CI Docker image and user Docker image.

Comment (yuqi1129) [reply dev/docker/trino/Dockerfile]: Modified, please check again.

File: dev/docker/trino/Dockerfile [hunk 2]
```diff
@@ -45,11 +45,18 @@ RUN rm -rf /usr/lib/trino/plugin/accumulo  \
...
     && rm -rf /usr/lib/trino/plugin/teradata-functions
-
 COPY --chown=trino:trino packages/trino/conf /etc/trino
 COPY --chown=trino:trino packages/gravitino-trino-connector/mysql-connector-java-8.0.27.jar /usr/lib/trino/plugin/iceberg/
 COPY --chown=trino:trino packages/gravitino-trino-connector/postgresql-42.7.0.jar /usr/lib/trino/plugin/iceberg/
-COPY --chown=trino:trino packages/gravitino-trino-connector /usr/lib/trino/plugin/gravitino
+
+ARG IMAGE_NAME
+RUN if [ "$IMAGE_NAME" = "datastrato/trino" ] ; then echo "Copying files for ${IMAGE_NAME} image" \
+  && COPY --chown=trino:trino packages/gravitino-trino-connector/mysql-connector-java-8.0.27.jar /usr/lib/trino/plugin/gravitino/ \
```
Comment (qqqttt123) [inline dev/docker/trino/Dockerfile:54]: For the user image, we should include the gravitino trinio connector.

Comment (yuqi1129) [reply dev/docker/trino/Dockerfile:54]: I have noticed this and I am considering how to address it.

File: dev/docker/build-docker.sh [hunk 3]
```diff
@@ -88,14 +88,14 @@ fi
 cd ${script_dir}/${component_type}
 if [[ "${platform_type}" == "all" ]]; then
   if [ ${build_latest} -eq 1 ]; then
-    docker buildx build --no-cache --pull --platform=linux/amd64,linux/arm64 ${build_args} --push --progress plain -f Dockerfile -t ${image_name}:latest -t ${image_name}:${tag_name} .
+    docker buildx build --build-arg IMAGE_NAME=${image_name} --no-cache --pull --platform=linux/amd64,linux/arm64 ${build_args} --push --progress plain -f Dockerfile -t ${image_name}:latest -t ${image_name}:${tag_name} .
```
Comment (xunliu) [inline dev/docker/build-docker.sh:91]: I think we can put `IMAGE_NAME=${image_name}` into `build_args` environment variable.

File: dev/docker/trino/Dockerfile [hunk 4]
```diff
@@ -45,11 +45,19 @@ RUN rm -rf /usr/lib/trino/plugin/accumulo  \
...
 COPY --chown=trino:trino packages/trino/conf /etc/trino
 COPY --chown=trino:trino packages/gravitino-trino-connector/mysql-connector-java-8.0.27.jar /usr/lib/trino/plugin/iceberg/
 COPY --chown=trino:trino packages/gravitino-trino-connector/postgresql-42.7.0.jar /usr/lib/trino/plugin/iceberg/
-COPY --chown=trino:trino packages/gravitino-trino-connector /usr/lib/trino/plugin/gravitino
+COPY --chown=trino:trino packages/gravitino-trino-connector /tmp/
+
+ARG IMAGE_NAME
+RUN if [ "$IMAGE_NAME" = "datastrato/trino" ] ; then \
+    mkdir /usr/lib/trino/plugin/gravitino; \
+    cp -rf /tmp/*.jar /usr/lib/trino/plugin/gravitino/; \
```
Comment (xunliu) [inline dev/docker/trino/Dockerfile:56]: I think would be better to change to `cp -r /tmp/gravitino-trino-connector /usr/lib/trino/plugin/gravitino`?

**Comment by jerryshao** [comment]
@xunliu please help to review this PR, thanks.

File: dev/docker/build-docker.sh (hunk 3 above)
Comment (yuqi1129) [reply dev/docker/build-docker.sh]: OK

File: dev/docker/trino/Dockerfile (hunk 4 above)
Comment (yuqi1129) [reply dev/docker/trino/Dockerfile]: I change command
`COPY --chown=trino:trino packages/gravitino-trino-connector /tmp/` to `COPY --chown=trino:trino packages/gravitino-trino-connector /tmp/gravitino-trino-connector`

and 
`cp -rf /tmp/*.jar /usr/lib/trino/plugin/gravitino/;` to `cp -rf /tmp/gravitino-trino-connector/*.jar /usr/lib/trino/plugin/gravitino/`

File: dev/docker/build-docker.sh (hunk 3 above)
Comment (yuqi1129) [reply dev/docker/build-docker.sh]: done

**Comment by jerryshao** [comment]
Any further comment on this PR @xunliu ?

**Comment by yuqi1129** [comment]
@xunliu 
Please take a look again.

**Review by diqiu50 — APPROVED** [review]
LGTM

**Review by xunliu — APPROVED** [review]
LGTM

**Comment by xunliu** [comment]
> Any further comment on this PR @xunliu ?

@jerryshao Yes, This PR will also support `datastrato/trino` and `datastrato/gravitino-ci-trino` Docker images.

**Pull request merged**